### ISSUE TITLE
Support Python2 Unicode Objects

### ISSUE THREAD

**Issue by dylan171** [frib-high-level-controls/FLAME#8]
Some of the Python2 standard libraries return unicode objects instead of str objects. Most often I have run into this when reading data from JSON data files and when parsing strings with regex groups.  However, when I pass these unicode strings within a FLAME configuration dictionary I get the following error "lists must contain only dict()s". It's easy to fix, just wrap everything in a "str()", but its annoying and error prone.

I have know idea how much work this is to implement on the C API level, but if it's not too difficult it would be a great improvement.

**Comment by mdavidsaver** [comment]
The "lists must contain only dict()s" error shouldn't come up with str vs. unicode.  Recursive Config is only supported via. a list.  For example:

```py
{'elements':[{'name':'elem1, ...}]}
```

So this error would be triggered by

```py
{'elements':{'name':'elem1, ...}}
```

Note the lack of square brackets.

**Comment by mdavidsaver** [comment]
As for str vs. unicode.  How should this be done?  Would encode as ascii be sufficient?

**Comment by dylan171** [comment]
I think that I have the data structure correct. 
```
Python 2.7.9 (default, Mar  1 2015, 12:57:24) 
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import flame
>>> flame.GLPSPrinter({ 'elements':[{ 'name':'drift_1', 'type':'drift' }] })
'drift_1: drift;\ndefault: LINE = (drift_1);\nUSE: default;\n'
>>> 
>>> flame.GLPSPrinter({ 'elements':[{ 'name':u'drift_1', 'type':u'drift' }] })
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: lists must contain only dict()s
```
This error surprised me too, but the python docs state that str and unicode are sequences of characters, so the call to PySequence_Check() must be true.

Ideally unicode would be fully supported, but I think that encoding as ASCII would be sufficient.

When a lattice file is parsed there is likely already an assumption that the file is plain ASCII.

**Comment by mdavidsaver** [comment]
> When a lattice file is parsed there is likely already an assumption that the file is plain ASCII.

Partly.  in the GLPS grammar keywords follow C89 identifier rules.  String values are arbitrary C strings.  So utf-8 is possible.

**Comment by mdavidsaver** [comment]
Resolved with 1c91b54d898054ab9d4930981599ec1cb1977657

I realized/remembered that there is a PyUnicode_AsUTF8String() which can be used in place of PyUnicode_AsASCIIString().
